src/Misc/layoutroot/run.sh

```diff
@@ -37,32 +37,55 @@ DIR="$( cd -P "$( dirname "$SOURCE" )" && pwd )"
 
 # Run
 shopt -s nocasematch
-if [[ "$1" == "localRun" ]]; then
-    "$DIR"/bin/Agent.Listener $*
-else
-    "$DIR"/bin/Agent.Listener run $*
 
-    # Return code 4 means the run once agent received an update message.
-    # Sleep at least 5 seconds (to allow the update process to start) and
-    # at most 20 seconds (to allow it to finish) then run the new agent
-    # again.
-    returnCode=$?
-    if [[ $returnCode == 4 ]]; then
-        delay 5
+# Determining if the "--once" flag was passed
+ONCE=false
+for a in $*; do
+    if [[ "$a" == "--once" ]]; then
+        ONCE=true
+    fi
+done
 
-        retry=0
-        while [[ $retry != 15 ]] && [ ! -x "$DIR"/bin/Agent.Listener ]; do
-            delay 1
-            retry=$[retry+1]
-        done
+FIRST_ARG=$1
+ARGUMENTS=$*
 
-        if [ ! -x "$DIR"/bin/Agent.Listener ]; then
-            echo "Failed to update within 20 seconds." >&2
-            exit 1
-        fi
-        
-        "$DIR"/bin/Agent.Listener run $*
+function runAgent {
+    if [[ "$FIRST_ARG" == "localRun" ]]; then
+        "$DIR"/bin/Agent.Listener
```

Comment:
Should the $ARGUMENTS be passed here as well?

Reply:
Fixed, Thanks